// rmc-cascader/lib/Cascader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _reactNative = require('react-native');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Picker = require('rmc-picker/lib/Picker');

var _Picker2 = _interopRequireDefault(_Picker);

var _CascaderMixin = require('./CascaderMixin');

var _CascaderMixin2 = _interopRequireDefault(_CascaderMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var styles = _reactNative.StyleSheet.create({
    item: {
        flex: 1
    },
    root: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingTop: 10,
        paddingBottom: 10
    }
});
var Cascader = _react2["default"].createClass({
    displayName: 'Cascader',

    mixins: [_CascaderMixin2["default"]],
    render: function render() {
        var _this = this;

        var value = this.state.value;
        var childrenTree = this.getChildrenTree();
        var cols = this.getColArray().map(function (_, i) {
            return _react2["default"].createElement(_reactNative.View, { style: styles.item, key: i }, _react2["default"].createElement(_Picker2["default"], { itemStyle: _this.props.pickerItemStyle, pure: false, selectedValue: value[i], onValueChange: function onValueChange(v) {
                    _this.onValueChange(i, v);
                } }, childrenTree[i] || []));
        });
        return _react2["default"].createElement(_reactNative.View, { style: styles.root }, cols);
    }
});
exports["default"] = Cascader;
module.exports = exports['default'];
