// antd-mobile/lib/toast/demo/basic.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=undefined;var _button=require('antd-mobile/components/button');var _button2=_interopRequireDefault(_button);var _whiteSpace=require('antd-mobile/components/white-space');var _whiteSpace2=_interopRequireDefault(_whiteSpace);var _wingBlank=require('antd-mobile/components/wing-blank');var _wingBlank2=_interopRequireDefault(_wingBlank);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _toast=require('antd-mobile/components/toast');var _toast2=_interopRequireDefault(_toast);var _react=require('react');var _react2=_interopRequireDefault(_react);

var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}
function showToast(){
_toast2["default"].info('这是一个 toast 提示!!!');
}
function successToast(){
_toast2["default"].success('加载成功!!!',2);
}
function failToast(){
_toast2["default"].fail('加载失败!!!');
}
function offline(){
_toast2["default"].offline('网络连接失败!!!');
}
function loadingToast(){
_toast2["default"].loading('加载中...',function(){
console.log('加载完成!!!');
});
}var
ToastExample=function(_React$Component){(0,_inherits3["default"])(ToastExample,_React$Component);function ToastExample(){(0,_classCallCheck3["default"])(this,ToastExample);return(0,_possibleConstructorReturn3["default"])(this,_React$Component.apply(this,arguments));}ToastExample.prototype.
componentDidMount=function componentDidMount(){
_reactNative.DeviceEventEmitter.addListener('navigatorBack',function(){
_toast2["default"].hide();
});
};ToastExample.prototype.
componentWillUnmount=function componentWillUnmount(){
_reactNative.DeviceEventEmitter.removeAllListeners('navigatorBack');
};ToastExample.prototype.
render=function render(){
return _react2["default"].createElement(_wingBlank2["default"],{style:{marginTop:30}},
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:showToast},"纯文字 toast"),
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:successToast},"成功 toast"),
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:failToast},"失败 toast"),
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:offline},"网络 toast"),
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:loadingToast},"加载中 toast"),
_react2["default"].createElement(_whiteSpace2["default"],null));
};return ToastExample;}(_react2["default"].Component);exports["default"]=ToastExample;module.exports=exports['default'];
