// antd-mobile/lib/toast/demo/basic.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=undefined;var _wingBlank=require('antd-mobile/components/wing-blank');var _wingBlank2=_interopRequireDefault(_wingBlank);var _button=require('antd-mobile/components/button');var _button2=_interopRequireDefault(_button);var _whiteSpace=require('antd-mobile/components/white-space');var _whiteSpace2=_interopRequireDefault(_whiteSpace);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _toast=require('antd-mobile/components/toast');var _toast2=_interopRequireDefault(_toast);var _jsxFileName='/Users/temupsc/git/antd/ant-design-mobile/components/toast/demo/basic.jsx';
var _react=require('react');var _react2=_interopRequireDefault(_react);

var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}
function showToast(){
_toast2["default"].info('这是一个 toast 提示!!!');
}
function successToast(){
_toast2["default"].success('加载成功!!!',1);
}
function showToastNoMask(){
_toast2["default"].info('无 mask 的 toast !!!',1,null,false);
}
function failToast(){
_toast2["default"].fail('加载失败!!!');
}
function offline(){
_toast2["default"].offline('网络连接失败!!!');
}
function loadingToast(){
_toast2["default"].loading('加载中...',1,function(){
console.log('加载完成!!!');
});
}var
ToastExample=function(_React$Component){(0,_inherits3["default"])(ToastExample,_React$Component);
function ToastExample(){(0,_classCallCheck3["default"])(this,ToastExample);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.apply(this,arguments));
_this.alwaysShowToast=function(){
_toast2["default"].info('这是一个 duration 为 0 的 toast!!!',0);
_this.timer=setTimeout(function(){
_toast2["default"].hide();
},5000);
};return _this;
}ToastExample.prototype.
componentDidMount=function componentDidMount(){
_reactNative.DeviceEventEmitter.addListener('navigatorBack',function(){
_toast2["default"].hide();
});
};ToastExample.prototype.
componentWillUnmount=function componentWillUnmount(){
_reactNative.DeviceEventEmitter.removeAllListeners('navigatorBack');
if(this.timer){
clearTimeout(this.timer);
this.timer=null;
}
};ToastExample.prototype.
render=function render(){
return _react2["default"].createElement(_wingBlank2["default"],{style:{marginTop:80},__source:{fileName:_jsxFileName,lineNumber:48}},
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:49}}),
_react2["default"].createElement(_button2["default"],{onClick:showToastNoMask,__source:{fileName:_jsxFileName,lineNumber:50}},'\u65E0 mask'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:51}}),
_react2["default"].createElement(_button2["default"],{onClick:showToast,__source:{fileName:_jsxFileName,lineNumber:52}},'\u7EAF\u6587\u5B57 toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:53}}),
_react2["default"].createElement(_button2["default"],{onClick:successToast,__source:{fileName:_jsxFileName,lineNumber:54}},'\u6210\u529F toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:55}}),
_react2["default"].createElement(_button2["default"],{onClick:failToast,__source:{fileName:_jsxFileName,lineNumber:56}},'\u5931\u8D25 toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:57}}),
_react2["default"].createElement(_button2["default"],{onClick:offline,__source:{fileName:_jsxFileName,lineNumber:58}},'\u7F51\u7EDC toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:59}}),
_react2["default"].createElement(_button2["default"],{onClick:loadingToast,__source:{fileName:_jsxFileName,lineNumber:60}},'\u52A0\u8F7D\u4E2D toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:61}}),
_react2["default"].createElement(_button2["default"],{onClick:this.alwaysShowToast,__source:{fileName:_jsxFileName,lineNumber:62}},'duration \u4E3A 0 \u7684 toast'),
_react2["default"].createElement(_whiteSpace2["default"],{__source:{fileName:_jsxFileName,lineNumber:63}}));

};return ToastExample;}(_react2["default"].Component);exports["default"]=ToastExample;module.exports=exports['default'];
