// antd-mobile/lib/action-sheet/demo/basic.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports.description=exports.title=undefined;var _actionSheet=require('antd-mobile/components/action-sheet');var _actionSheet2=_interopRequireDefault(_actionSheet);var _button=require('antd-mobile/components/button');var _button2=_interopRequireDefault(_button);var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}

var BUTTONS=['操作一','操作二','操作三','删除','取消'];
var DESTRUCTIVE_INDEX=3;
var CANCEL_INDEX=4;exports["default"]=
_react2["default"].createClass({displayName:'basic',
getInitialState:function getInitialState(){
return{
clicked:'none',
text:''};

},
render:function render(){
return _react2["default"].createElement(_reactNative.View,{style:{marginTop:30}},
_react2["default"].createElement(_reactNative.View,{style:[{padding:8}]},
_react2["default"].createElement(_button2["default"],{onClick:this.showActionSheet},"默认状态操作列表")),

_react2["default"].createElement(_reactNative.Text,{style:[{padding:8}]},
"点击过的按钮: ",
this.state.clicked),
_react2["default"].createElement(_reactNative.View,{style:[{padding:8}]},
_react2["default"].createElement(_button2["default"],{onClick:this.showShareActionSheet},"带分享功能的操作列表")),

_react2["default"].createElement(_reactNative.Text,{style:[{padding:8}]},this.state.text));
},
showActionSheet:function showActionSheet(){var _this=this;
_actionSheet2["default"].showActionSheetWithOptions({
title:'标题',
message:'我是描述我是描述',
options:BUTTONS,
cancelButtonIndex:CANCEL_INDEX,
destructiveButtonIndex:DESTRUCTIVE_INDEX},
function(buttonIndex){
_this.setState({clicked:BUTTONS[buttonIndex]});
});
},
showShareActionSheet:function showShareActionSheet(){var _this2=this;
var opts={
url:'https://www.alipay.com/',
message:'message to go with the shared url',
excludedActivityTypes:[
_react2["default"].createElement(_button2["default"],{onClick:function onClick(){return _actionSheet2["default"].close();}},"close ActionSheet")],

subject:null};

if(_reactNative.Platform.OS==='ios'){
opts.subject='a subject to go in the email heading';
opts.excludedActivityTypes=[
'com.apple.UIKit.activity.PostToTwitter'];

}
_actionSheet2["default"].showShareActionSheetWithOptions(opts,function(error){return alert(error);},function(success,method){
var text=void 0;
if(success){
text='\u901A\u8FC7 '+method+' \u5206\u4EAB';
}else
{
text='您没有分享';
}
_this2.setState({text:text});
});
}});

var title=exports.title='ActionSheet';
var description=exports.description='ActionSheet example';
