// antd-mobile/lib/image-picker/index.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=undefined;var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _jsxFileName='/Users/temupsc/git/antd/ant-design-mobile/components/image-picker/index.jsx';var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');


var _style=require('./style/');var _style2=_interopRequireDefault(_style);
var _ImageRoll=require('./ImageRoll');var _ImageRoll2=_interopRequireDefault(_ImageRoll);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}var
ImagePicker=function(_React$Component){(0,_inherits3["default"])(ImagePicker,_React$Component);
function ImagePicker(props){(0,_classCallCheck3["default"])(this,ImagePicker);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.call(this,props));
_this.onPressIn=function(){
var styles=_this.props.styles;
_this.plusWrap.setNativeProps({
style:[styles.item,styles.size,styles.plusWrapHighlight]});

};
_this.onPressOut=function(){
var styles=_this.props.styles;
_this.plusWrap.setNativeProps({
style:[styles.item,styles.size,styles.plusWrapNormal]});

};















_this.showPicker=function(){



















_this.setState({
visible:true});


};
_this.hideImageRoll=function(){
_this.setState({
visible:false});

};
_this.state={
visible:false};return _this;

}ImagePicker.prototype.
addImage=function addImage(imageObj){
if(!imageObj.url){
imageObj.url=imageObj.uri;
delete imageObj.uri;
}var _props$files=
this.props.files,files=_props$files===undefined?[]:_props$files;
var newImages=files.concat(imageObj);
if(this.props.onChange){
this.props.onChange(newImages,'add');
}
};ImagePicker.prototype.
removeImage=function removeImage(idx){
var newImages=[];var _props$files2=
this.props.files,files=_props$files2===undefined?[]:_props$files2;
files.forEach(function(image,index){
if(index!==idx){
newImages.push(image);
}
});
if(this.props.onChange){
this.props.onChange(newImages,'remove',idx);
}
};ImagePicker.prototype.
render=function render(){var _this2=this;var _props=
this.props,styles=_props.styles,_props$files3=_props.files,files=_props$files3===undefined?[]:_props$files3;
var filesView=files.map(function(item,index){return _react2["default"].createElement(_reactNative.View,{key:index,style:[styles.item,styles.size],__source:{fileName:_jsxFileName,lineNumber:96}},
_react2["default"].createElement(_reactNative.Image,{source:{uri:item.url},style:[styles.size,styles.image],__source:{fileName:_jsxFileName,lineNumber:97}}),
_react2["default"].createElement(_reactNative.TouchableWithoutFeedback,{onPress:function onPress(){return _this2.removeImage(index);},__source:{fileName:_jsxFileName,lineNumber:98}},
_react2["default"].createElement(_reactNative.View,{style:styles.closeWrap,__source:{fileName:_jsxFileName,lineNumber:99}},
_react2["default"].createElement(_reactNative.Text,{style:styles.closeText,__source:{fileName:_jsxFileName,lineNumber:100}},'\xD7'))));});



var imageRollEl=_react2["default"].createElement(_ImageRoll2["default"],{onCancel:this.hideImageRoll,onSelected:function onSelected(imgObj){return _this2.addImage(imgObj);},__source:{fileName:_jsxFileName,lineNumber:104}});
return _react2["default"].createElement(_reactNative.View,{style:styles.container,__source:{fileName:_jsxFileName,lineNumber:105}},
filesView,
_react2["default"].createElement(_reactNative.TouchableWithoutFeedback,{onPress:this.showPicker,onPressIn:this.onPressIn,onPressOut:this.onPressOut,__source:{fileName:_jsxFileName,lineNumber:107}},
_react2["default"].createElement(_reactNative.View,{ref:function ref(conponent){return _this2.plusWrap=conponent;},style:[styles.item,styles.size,styles.plusWrap,styles.plusWrapNormal],__source:{fileName:_jsxFileName,lineNumber:108}},
_react2["default"].createElement(_reactNative.Text,{style:[styles.plusNormal,styles.plusText],__source:{fileName:_jsxFileName,lineNumber:109}},'+'))),


this.state.visible?imageRollEl:null);

};return ImagePicker;}(_react2["default"].Component);exports["default"]=ImagePicker;

ImagePicker.defaultProps={
styles:_style2["default"],
onChange:function onChange(){},
files:[]};module.exports=exports['default'];
