// antd-mobile/lib/list-view/demo/basic-row.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports.description=exports.title=exports["default"]=undefined;var _listView=require('antd-mobile/components/list-view');var _listView2=_interopRequireDefault(_listView);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _jsxFileName='/Users/temupsc/git/antd/ant-design-mobile/components/list-view/demo/basic-row.jsx';
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');

var _objectAssign=require('object-assign');var _objectAssign2=_interopRequireDefault(_objectAssign);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}
var data=[
{
img:'https://zos.alipayobjects.com/rmsportal/dKbkpPXKfvZzWCM.png',
title:'Meet hotel',
des:'不是所有的兼职汪都需要风吹日晒'},

{
img:'https://zos.alipayobjects.com/rmsportal/XmwCzSeJiqpkuMB.png',
title:'McDonald\'s invites you',
des:'不是所有的兼职汪都需要风吹日晒'},

{
img:'https://zos.alipayobjects.com/rmsportal/hfVtzEhPzTUewPm.png',
title:'Eat the week',
des:'不是所有的兼职汪都需要风吹日晒'}];


var index=data.length-1;
var NUM_ROWS=20;
var pageIndex=0;var
BasicRowDemo=function(_React$Component){(0,_inherits3["default"])(BasicRowDemo,_React$Component);
function BasicRowDemo(props){(0,_classCallCheck3["default"])(this,BasicRowDemo);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.call(this,props));
_this.genData=function(){var pIndex=arguments.length>0&&arguments[0]!==undefined?arguments[0]:0;
var dataBlob={};
for(var i=0;i<NUM_ROWS;i++){
var ii=pIndex*NUM_ROWS+i;
dataBlob[''+ii]='row - '+ii;
}
return dataBlob;
};
_this.onEndReached=function(_event){


_this.setState({isLoading:true});
setTimeout(function(){
_this.rData=(0,_objectAssign2["default"])({},_this.rData,_this.genData(++pageIndex));
_this.setState({
dataSource:_this.state.dataSource.cloneWithRows(_this.rData),
isLoading:false});

},1000);
};
var dataSource=new _listView2["default"].DataSource({
rowHasChanged:function rowHasChanged(row1,row2){return row1!==row2;}});

_this.rData={};
_this.state={
dataSource:dataSource.cloneWithRows(_this.genData()),
isLoading:false};return _this;

}BasicRowDemo.prototype.
render=function render(){var _this2=this;
var separator=function separator(sectionID,rowID){return _react2["default"].createElement(_reactNative.View,{key:sectionID+'-'+rowID,style:{
backgroundColor:'#F5F5F9',
height:8,
borderStyle:'solid',
borderTopWidth:1,
borderTopColor:'#ECECED',
borderBottomWidth:1,
borderBottomColor:'#ECECED'},__source:{fileName:_jsxFileName,lineNumber:59}});};

var row=function row(_rowData,sectionID,rowID){var highlightRow=arguments.length>3&&arguments[3]!==undefined?arguments[3]:function(_sId,_rId){};
if(index<0){
index=data.length-1;
}
var obj=data[index--];
return _react2["default"].createElement(_reactNative.View,{key:rowID,__source:{fileName:_jsxFileName,lineNumber:73}},
_react2["default"].createElement(_reactNative.TouchableHighlight,{underlayColor:'rgba(100,100,100,0.2)',style:[{padding:8,backgroundColor:'white'}],onPress:function onPress(){highlightRow(sectionID,rowID);},__source:{fileName:_jsxFileName,lineNumber:74}},
_react2["default"].createElement(_reactNative.View,{__source:{fileName:_jsxFileName,lineNumber:75}},
_react2["default"].createElement(_reactNative.View,{style:[{marginBottom:8,borderStyle:'solid',borderBottomWidth:1,borderBottomColor:'#F6F6F6'}],__source:{fileName:_jsxFileName,lineNumber:76}},
_react2["default"].createElement(_reactNative.Text,{style:{fontSize:18,fontWeight:'500',padding:2},__source:{fileName:_jsxFileName,lineNumber:77}},obj.title)),

_react2["default"].createElement(_reactNative.View,{style:[{flexDirection:'row'}],__source:{fileName:_jsxFileName,lineNumber:79}},
_react2["default"].createElement(_reactNative.Image,{style:[{height:64,width:64,marginRight:8}],source:{uri:obj.img},__source:{fileName:_jsxFileName,lineNumber:80}}),
_react2["default"].createElement(_reactNative.View,{__source:{fileName:_jsxFileName,lineNumber:81}},
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:82}},obj.des,' - ',rowID),
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:83}},_this2.props.highlightRow),
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:84}},_react2["default"].createElement(_reactNative.Text,{style:[{fontSize:24,color:'#FF6E27'}],__source:{fileName:_jsxFileName,lineNumber:84}},'35'),'\xA5'))))));





};
var loadingTxt=this.state.isLoading?'Loading...':'Loaded';
return _react2["default"].createElement(_listView2["default"],{dataSource:this.state.dataSource,renderHeader:function renderHeader(){return _react2["default"].createElement(_reactNative.Text,{style:[{padding:8}],__source:{fileName:_jsxFileName,lineNumber:92}},'header');},renderFooter:function renderFooter(){return _react2["default"].createElement(_reactNative.Text,{style:{padding:30,textAlign:'center'},__source:{fileName:_jsxFileName,lineNumber:92}},' ',loadingTxt,' ');},renderRow:row,renderSeparator:separator,pageSize:4,scrollRenderAheadDistance:500,scrollEventThrottle:20,onEndReached:this.onEndReached,onEndReachedThreshold:10,__source:{fileName:_jsxFileName,lineNumber:92}});
};return BasicRowDemo;}(_react2["default"].Component);exports["default"]=BasicRowDemo;

var title=exports.title='ListView Row';
var description=exports.description='ListView Row example';
