// antd-mobile/lib/list-view/demo/basic.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports.description=exports.title=exports["default"]=undefined;var _listView=require('antd-mobile/components/list-view');var _listView2=_interopRequireDefault(_listView);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _jsxFileName='/Users/temupsc/git/antd/ant-design-mobile/components/list-view/demo/basic.jsx';
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}

var data=[
{
img:'https://zos.alipayobjects.com/rmsportal/dKbkpPXKfvZzWCM.png',
title:'Meet hotel',
des:'不是所有的兼职汪都需要风吹日晒'},

{
img:'https://zos.alipayobjects.com/rmsportal/XmwCzSeJiqpkuMB.png',
title:'McDonald\'s invites you',
des:'不是所有的兼职汪都需要风吹日晒'},

{
img:'https://zos.alipayobjects.com/rmsportal/hfVtzEhPzTUewPm.png',
title:'Eat the week',
des:'不是所有的兼职汪都需要风吹日晒'}];


var index=data.length-1;
var NUM_SECTIONS=5;
var NUM_ROWS_PER_SECTION=5;
var pageIndex=0;var
BasicDemo=function(_React$Component){(0,_inherits3["default"])(BasicDemo,_React$Component);
function BasicDemo(props){(0,_classCallCheck3["default"])(this,BasicDemo);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.call(this,props));
_this._genData=function(){var pIndex=arguments.length>0&&arguments[0]!==undefined?arguments[0]:0;
for(var i=0;i<NUM_SECTIONS;i++){
var ii=pIndex*NUM_SECTIONS+i;
var sectionName='Section '+ii;
_this.sectionIDs.push(sectionName);
_this.dataBlob[sectionName]=sectionName;
_this.rowIDs[ii]=[];
for(var jj=0;jj<NUM_ROWS_PER_SECTION;jj++){
var rowName='S'+ii+', R'+jj;
_this.rowIDs[ii].push(rowName);
_this.dataBlob[rowName]=rowName;
}
}

_this.sectionIDs=[].concat(_this.sectionIDs);
_this.rowIDs=[].concat(_this.rowIDs);
};
_this.onEndReached=function(_event){

_this.setState({isLoading:true});
setTimeout(function(){
_this._genData(++pageIndex);
_this.setState({
dataSource:_this.state.dataSource.cloneWithRowsAndSections(_this.dataBlob,_this.sectionIDs,_this.rowIDs),
isLoading:false});

},1000);
};
_this.renderSectionHeader=function(sectionData){
return _react2["default"].createElement(_reactNative.Text,{style:[{padding:10,backgroundColor:'rgba(255,255,255,0.8)'}],__source:{fileName:_jsxFileName,lineNumber:58}},'Task '+
sectionData.split(' ')[1]);

};
var getSectionData=function getSectionData(dataBlob,sectionID){return dataBlob[sectionID];};
var getRowData=function getRowData(dataBlob,_sectionID,rowID){return dataBlob[rowID];};
var dataSource=new _listView2["default"].DataSource({
getRowData:getRowData,
getSectionHeaderData:getSectionData,
rowHasChanged:function rowHasChanged(row1,row2){return row1!==row2;},
sectionHeaderHasChanged:function sectionHeaderHasChanged(s1,s2){return s1!==s2;}});

_this.dataBlob={};
_this.sectionIDs=[];
_this.rowIDs=[];
_this._genData();
_this.state={
dataSource:dataSource.cloneWithRowsAndSections(_this.dataBlob,_this.sectionIDs,_this.rowIDs),
headerPressCount:0};return _this;

}BasicDemo.prototype.
render=function render(){var _this2=this;
var separator=function separator(sectionID,rowID){return _react2["default"].createElement(_reactNative.View,{key:sectionID+'-'+rowID,style:{
backgroundColor:'#F5F5F9',
height:8,
borderStyle:'solid',
borderTopWidth:1,
borderTopColor:'#ECECED',
borderBottomWidth:1,
borderBottomColor:'#ECECED'},__source:{fileName:_jsxFileName,lineNumber:80}});};

var row=function row(_rowData,sectionID,rowID){var highlightRow=arguments.length>3&&arguments[3]!==undefined?arguments[3]:function(_sId,_rId){};
if(index<0){
index=data.length-1;
}
var obj=data[index--];
return _react2["default"].createElement(_reactNative.View,{key:rowID,__source:{fileName:_jsxFileName,lineNumber:94}},
_react2["default"].createElement(_reactNative.TouchableHighlight,{underlayColor:'rgba(100,100,100,0.2)',style:[{padding:8,backgroundColor:'white'}],onPress:function onPress(){highlightRow(sectionID,rowID);},__source:{fileName:_jsxFileName,lineNumber:95}},
_react2["default"].createElement(_reactNative.View,{__source:{fileName:_jsxFileName,lineNumber:96}},
_react2["default"].createElement(_reactNative.View,{style:[{marginBottom:8,borderStyle:'solid',borderBottomWidth:1,borderBottomColor:'#F6F6F6'}],__source:{fileName:_jsxFileName,lineNumber:97}},
_react2["default"].createElement(_reactNative.Text,{style:{fontSize:18,fontWeight:'500',padding:2},__source:{fileName:_jsxFileName,lineNumber:98}},obj.title)),

_react2["default"].createElement(_reactNative.View,{style:[{flexDirection:'row'}],__source:{fileName:_jsxFileName,lineNumber:100}},
_react2["default"].createElement(_reactNative.Image,{style:[{height:64,width:64,marginRight:8}],source:{uri:obj.img},__source:{fileName:_jsxFileName,lineNumber:101}}),
_react2["default"].createElement(_reactNative.View,{__source:{fileName:_jsxFileName,lineNumber:102}},
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:103}},obj.des,' - ',rowID),
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:104}},_this2.props.highlightRow),
_react2["default"].createElement(_reactNative.Text,{__source:{fileName:_jsxFileName,lineNumber:105}},_react2["default"].createElement(_reactNative.Text,{style:[{fontSize:24,color:'#FF6E27'}],__source:{fileName:_jsxFileName,lineNumber:105}},'35'),'\xA5'))))));





};
var loadingTxt=this.state.isLoading?'Loading...':'Loaded';
return _react2["default"].createElement(_listView2["default"],{dataSource:this.state.dataSource,renderHeader:function renderHeader(){return _react2["default"].createElement(_reactNative.Text,{style:[{padding:8}],__source:{fileName:_jsxFileName,lineNumber:113}},'header');},renderFooter:function renderFooter(){return _react2["default"].createElement(_reactNative.Text,{style:[{padding:30,textAlign:'center'}],__source:{fileName:_jsxFileName,lineNumber:113}},' ',loadingTxt,' ');},renderSectionHeader:this.renderSectionHeader,renderRow:row,renderSeparator:separator,pageSize:4,onEndReached:this.onEndReached,onEndReachedThreshold:10,__source:{fileName:_jsxFileName,lineNumber:113}});
};return BasicDemo;}(_react2["default"].Component);exports["default"]=BasicDemo;

var title=exports.title='ListView';
var description=exports.description='ListView example';
