// @boluome/oto_saas_web_app_component/webpack.config.babel.js
import path    from 'path'
import webpack from 'webpack'
import htmlWebpackPlugin from 'html-webpack-plugin'
import pxtorem from 'postcss-pxtorem'
import autoprefixer from 'autoprefixer'
const bProdEnv    = process.env.NODE_ENV === 'online'

const ROOT_PATH   = path.resolve(__dirname)
const DEMO_PATH   = path.resolve(ROOT_PATH, 'demo')
const SASS_PATH   = path.resolve(ROOT_PATH, 'sass')
const SRC_PATH    = path.resolve(ROOT_PATH, 'src')
const DIST_PATH   = path.resolve(ROOT_PATH, 'dist')
const MODULE_PATH = path.resolve(ROOT_PATH, 'node_modules')
const ASSET_PATH  = path.resolve(ROOT_PATH, 'assets')

const config = {
  entry: {
    'app': DEMO_PATH,
    'vendor': ['react', 'react-dom', 'react-router', 'react-redux', 'redux']
  },
  output: {
    path         : DIST_PATH,
    filename     : 'bundle.js?[hash]',
    chunkFilename: '[name].chunk.js?[hash]',
    publicPath   : '/'
  },
  plugins: [
    new htmlWebpackPlugin({
      template: path.resolve(ASSET_PATH, 'index.html'),
      inject  : true
    }),
    new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.bundle.js?[hash]')
  ],
  module: {
    loaders: [
      {
        test: /\.jsx?$/,
        loader: 'babel',
        include: [ SRC_PATH, DEMO_PATH, MODULE_PATH ]
      },
      {
        test: /\.(css|scss|sass)$/,
        loaders: ['style', 'css', 'postcss', 'sass'],
        include: [ SRC_PATH, DEMO_PATH, MODULE_PATH, SASS_PATH ]
      },
      {
        test: /\.(png|jpg|gif|svg)$/,
        include: [ SRC_PATH, MODULE_PATH ],
        loader: 'url',
        query : {
          limit: 10000,
          name : '[name].[ext]?[hash]'
        }
      }
    ]
  },
  postcss: [
    autoprefixer,
    pxtorem({
      rootValue: 100,
      propWhiteList: [],
    })
  ],
  resolve: {
    modulesDirectories: [
      'node_modules',
      'src'
    ],
    alias: {
      '@boluome/oto_saas_web_app_component': path.resolve(SRC_PATH),
      'common-sass': path.resolve(ROOT_PATH, 'sass')
    }
  }
}
console.log('env: ', bProdEnv)
if(!bProdEnv) {
  config.devServer = {
    hot: true,
    inline: true,
    historyApiFallback: true,
    port: '9003',
    host: '0.0.0.0'
  }
  config.plugins = config.plugins.concat([
    new webpack.HotModuleReplacementPlugin()
  ])
} else {
  config.plugins = config.plugins.concat([
    new webpack.DefinePlugin({
      'process.env': {
        NODE_ENV: JSON.parse('production')
      }
    }),
    new webpack.optimize.UglifyJsPlugin({
      compress: {
        warnings: false
      }
    })
  ])
}

export default config
