// antd-mobile/lib/popup/demo/basic.js
'use strict';Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=undefined;var _whiteSpace=require('antd-mobile/components/white-space');var _whiteSpace2=_interopRequireDefault(_whiteSpace);var _popup=require('antd-mobile/components/popup');var _popup2=_interopRequireDefault(_popup);var _defineProperty2=require('babel-runtime/helpers/defineProperty');var _defineProperty3=_interopRequireDefault(_defineProperty2);var _button=require('antd-mobile/components/button');var _button2=_interopRequireDefault(_button);var _list=require('antd-mobile/components/list');var _list2=_interopRequireDefault(_list);var _classCallCheck2=require('babel-runtime/helpers/classCallCheck');var _classCallCheck3=_interopRequireDefault(_classCallCheck2);var _possibleConstructorReturn2=require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3=_interopRequireDefault(_possibleConstructorReturn2);var _inherits2=require('babel-runtime/helpers/inherits');var _inherits3=_interopRequireDefault(_inherits2);var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj};}var

PopupExample=function(_React$Component){(0,_inherits3["default"])(PopupExample,_React$Component);
function PopupExample(props){(0,_classCallCheck3["default"])(this,PopupExample);var _this=(0,_possibleConstructorReturn3["default"])(this,
_React$Component.call(this,props));
_this.getPopupContent=function(num){
return _react2["default"].createElement(_reactNative.View,null,
_react2["default"].createElement(_list2["default"],{renderHeader:function renderHeader(){return'委托买入';}},
_react2["default"].createElement(_list2["default"].Item,null,"股票名称"),
_react2["default"].createElement(_list2["default"].Item,null,"股票代码"),
_react2["default"].createElement(_list2["default"].Item,null,"买入价格"),
_react2["default"].createElement(_list2["default"].Item,null,"买入数量")),
_react2["default"].createElement(_reactNative.View,{style:{padding:10}},
_react2["default"].createElement(_reactNative.Text,{style:{color:'gray'}},"投资说明投资说明..."),
_react2["default"].createElement(_reactNative.Text,{style:{color:'gray'}},"交易金额以实际成交为准")),
_react2["default"].createElement(_reactNative.View,{style:{padding:6}},
_react2["default"].createElement(_button2["default"],{type:"primary",onClick:function onClick(){return _this.onClose('cancel',num);}},"买入")));

};
_this.state={
sel0:'',
sel1:''};return _this;

}PopupExample.prototype.
onClose=function onClose(sel,num){
this.setState((0,_defineProperty3["default"])({},'sel'+num,sel));
_popup2["default"].hide();
};PopupExample.prototype.
render=function render(){var _this2=this;
return _react2["default"].createElement(_reactNative.View,{style:{marginTop:30,paddingHorizontal:30}},
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:function onClick(){return _popup2["default"].show(_this2.getPopupContent(0));}},"向下弹出效果"),
_react2["default"].createElement(_whiteSpace2["default"],null),
_react2["default"].createElement(_button2["default"],{onClick:function onClick(){return _popup2["default"].show(_this2.getPopupContent(1),{
maskClosable:true,
animationType:'slide-up',
onMaskClose:function onMaskClose(){return new Promise(function(resolve){setTimeout(resolve,200);});}});}},
"向上弹出效果"));
};return PopupExample;}(_react2["default"].Component);exports["default"]=PopupExample;module.exports=exports['default'];
